### DISCUSSION TITLE
Any chance to use less memory ?

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#3913] [CP-SAT questions]
Hello,
  my model contains a lot of kLinear2 constraints representing disjunctive constraints. With 8 search workers and about 10 million kLinear2 constraints memory usage is around 60gb during calculation.
  Since such high number of constraints can be generally unexpected is there a chance to use considerable less memory in a future ?

Thanks,
Jan

Proto file was over limit of github, it can be dowloaded via following link https://www.uschovna.cz/en/zasilka/ML4MHVYFNY5NMMXK-5N8

**Comment by lperron** [comment]
I downloaded the proto. Quite challenging :-)

We have reduced a bit memory on main for large linear. Linear2 and fairly optimized and stored in a differential logic engine.
